src/plugins/data/common/search/search_source/create_search_source.ts

```diff
@@ -56,6 +57,18 @@ export const createSearchSource = (
 ) => async (searchSourceFields: SearchSourceFields = {}) => {
   const fields = { ...searchSourceFields };
 
+  // When we load a saved search and the saved search contains a non index pattern data source this step creates the temperary index patterns and sets the appriopriate query
+  if (
+    searchSourceDependencies.queryStringService &&
+    fields.query?.dataset &&
+    fields.query?.dataset?.type !== 'INDEX_PATTERN' &&
+    !indexPatterns.isPresentInCache(fields.query.dataset.id)
+  ) {
+    await searchSourceDependencies.queryStringService
+      .getDatasetService()
+      .cacheDataset(fields.query?.dataset);
```

Comment:
I assume we arent solving the problem of caching in multiple places today right?

Reply:
Yup. Adding these can increase the scope of the PR, rather I have created an issues (https://github.com/opensearch-project/OpenSearch-Dashboards/issues/8496) that we can tackle separately